tests/modules/images/webp-uploads/load-tests.php

```diff
@@ -484,6 +484,61 @@ public function data_provider_not_supported_webp_images() {
 		yield 'GIFT image' => array( TESTS_PLUGIN_DIR . '/tests/testdata/modules/images/earth.gif' );
 	}
 
+	/**
+	 * Ensure jpg image is replaced if the request is for the frontend context.
+	 *
+	 * @group webp_uploads_update_image_references
+	 *
+	 * @test
+	 */
+	public function it_should_not_replace_image_references_when_request_is_for_frontend_context() {
+		$this->mock_action_template_redirect();
+
+		$attachment_id = $this->factory->attachment->create_upload_object(
+			TESTS_PLUGIN_DIR . '/tests/testdata/modules/images/leafs.jpg'
+		);
+
+		$tag = wp_get_attachment_image( $attachment_id, 'medium', false, array( 'class' => "wp-image-{$attachment_id}" ) );
+		$this->assertNotSame( $tag, webp_uploads_update_image_references( $tag ) );
+	}
+
+	/**
+	 * Prevent replacing a jpg image if the request is not for the frontend context.
+	 * In this test case, the request if for the `feed` page.
+	 *
+	 * @group webp_uploads_update_image_references
+	 *
+	 * @test
+	 */
+	public function it_should_not_replace_image_references_when_request_is_not_for_frontend_context() {
+		$this->mock_action_template_redirect();
+		$this->mock_is_feed_page();
```

Comment:
This isn't a real scenario so I don't think we should test it. `template_redirect` will never fire if it is a feed. Let's rather focus on purely the feed check in this test.
```suggestion
		$this->mock_is_feed_page();
```

Reply:
@felixarntz While testing this, `template_redirect` did fire for `rss2` feeds. You can verify this by creating a test post with images and check the `rss2` page `http://localhost:8888/rss2`

Reply:
@akshitsethi Oh sorry, I didn't know that, you're right. In that case this is good to keep.

That is another good reminder though why we need to also check for `did_action( 'wp_head' )`.